File: Implementation/Apple-and-Orange/AppleAndOrange.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution {

    public static void main(String[] args) {
    Scanner in = new Scanner(System.in);

		int s = in.nextInt();
		int t = in.nextInt();

		int min = Math.min(s, t);
		int max = Math.max(s, t);

		int a = in.nextInt(); // apple tree
		int b = in.nextInt(); // orange tree

		int m = in.nextInt(); // number of apples
		int n = in.nextInt(); // number of oranges

		// apples
		int apples = 0;
		for (int i = 0; i < m; i++) {
			int k = in.nextInt() + a;
			if (k >= min && k <= max) {
				apples++;
			}
		}

		// oranges
		int oranges = 0;
		for (int i = 0; i < n; i++) {
			int k = in.nextInt() + b;
			if (k >= min && k <= max) {
				oranges++;
			}
		}
		
		System.out.println(apples);
		System.out.println(oranges);
	}
}

File: Algorithms/Implementation/Apple and Orange/appleAndOrange.py
s,t = input().strip().split(' ')
s,t = [int(s),int(t)]
a,b = input().strip().split(' ')
a,b = [int(a),int(b)]
m,n = input().strip().split(' ')
m,n = [int(m),int(n)]
apples = [int(apple_temp) for apple_temp in input().strip().split(' ')]
oranges = [int(orange_temp) for orange_temp in input().strip().split(' ')]

min = min(s, t)
max = max(s, t)

app = 0
orr = 0
for apple in apples:
    k = apple + a
    if k >= min and k <= max:
        app = app + 1
    
for orange in oranges:
    k = orange + b
    if k >= min and k <= max:
        orr = orr + 1
    
print(app)
print(orr)
